File: src/main/java/Piruru/cards/AllForOneButSkills.java
package Piruru.cards;

import Piruru.abstracts.PiruruCard;
import Piruru.actions.AllForSkills;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;



public class AllForOneButSkills extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(AllForOneButSkills.class));
    private static final int COST = 3;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.RARE;
    private static final CardTarget TARGET = CardTarget.SELF;

    public AllForOneButSkills() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST);
        isEthereal = true;
        exhaust = true;
    }

    @Override
    public void upgrade() {
        if (!upgraded) {
            upgradeName();
            rawDescription = cardStrings.UPGRADE_DESCRIPTION;
            initializeDescription();
        }
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new AllForSkills(upgraded));
    }
}

File: src/main/kotlin/Piruru/cards/AllForOneButSkills.kt
package Piruru.cards

import Piruru.Piruluk.Statics.makeID
import Piruru.abstracts.PiruruCard
import Piruru.actions.AllForSkills
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster


class AllForOneButSkills : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST) {
    init {
        isEthereal = true
        exhaust = true
    }

    override fun upgrade() {
        if (!upgraded) {
            upgradeName()
            rawDescription = cardStrings.UPGRADE_DESCRIPTION
            initializeDescription()
        }
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        act(AllForSkills(upgraded))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(AllForOneButSkills::class.java))
        private val COST = 3
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.RARE
        private val TARGET = AbstractCard.CardTarget.SELF
    }
}
